content/src/controller.js

```diff
@@ -2009,6 +2018,55 @@ function sortByName(a, b) {
 function renderDirectory(position) {
   var pane = paneatpos(position);
   var mpp = model.pane[pane];
+  mpp.bydate = defaultDirSortingByDate();
+  var filename = mpp.filename;
+  view.setPaneLinkText(pane, getUpdatedLinksArray(pane), filename, model.ownername);
+  updateTopControls(false);
+}
+
+
+function updateSortResults(pane) {
+  view.setPaneLinks(pane,getUpdatedLinksArray(pane));
+  updateTopControls(false);
+}
+
+function updateSearchResults(pane, search, cb) {
+  var mpp = model.pane[pane];
+  if(specialowner()) {
```

Comment:
No, search is there for others too. But, it fetch from the server only if the special owner. Otherwise It will do the searching locally. we can assume that one user will not have more than 600 files. 

And also, since the time taken for fetching is quite considerable even in my machine, I maintained a simple cache for search results.